app/controllers/api/v1/jobs/messages_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Api::V1::Jobs::MessagesController < ApplicationController
+  swagger_controller :messages, 'Messages'
+
+  swagger_api :create do
+    summary 'Send a message to a user'
+    notes 'Create a new message. business_name is required for Requester but not for Professional.'
+  end
+
+  def create
+    job = Job.find(params[:job_id])
+    sender = User.find_by_api_key(params[:api_key])
+    requester = job.requester
+    if requester == sender
+      recipient = User.find_by_business_name(params[:business_name])
+    else
+      recipient = requester
+    end
+    @name = if recipient.role == "professional"
+              recipient.business_name
+            else
+              recipient.full_name
+            end
+    Message.create!(body: params[:body], sender: sender, recipient: recipient, job: job)
+    render status: 201
+  end
+
+  def index
+    @messages = Message.all
```

Comment:
I think this should look more like `@messages = Message.by_job(params[:job_id])`, so we scope to the job the user is querying on. Currently this will give everyone's messages